overlord/devicestate/devicestate.go

```diff
@@ -293,6 +293,26 @@ func CanManageRefreshes(st *state.State) bool {
 	return false
 }
 
+// splitTsAfterEdge takes a taskset ts and splits it into two tasksets
+// after the first occurrence of the given task kind
+func splitTsAfterEdge(edge state.TaskSetEdge, ts *state.TaskSet) (*state.TaskSet, *state.TaskSet) {
```

Comment:
this seems doesn't need to be a split anymore but could return just the relevant 4 tasks,  otoh unclear we need it at all if the original code marked more edges about starts and ends etc

Reply:
Indeed, I reworked the code to not use indexing anymore